lib/ui/contact-list.dart

```diff
@@ -0,0 +1,96 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_contact_list/ui/contact-detail.dart';
+
+//fake class and data
+//will remove later 
+class Contact{
+    int id;
+    String firstName; 
+    String lastName; 
+    String phoneNumber;
+
+    Contact({required this.id, required this.firstName, required this.lastName, required this.phoneNumber}); 
+
+    @override 
+    String toString() {
+        return("id: $id\nFirst name: $firstName\nLast name: $lastName\nTel: $phoneNumber"); 
+  }
+}
+
+class ContactList extends StatefulWidget {
+    const ContactList({ Key? key }) : super(key: key);
+
+    @override
+    _ContactListState createState() => _ContactListState();
+}
+
+class _ContactListState extends State<ContactList> { 
+    //create some fake data and assign to contactList 
+    List<Contact> contactList = List.generate(10, (int index) => Contact(id: index, firstName: "First ", lastName: "Last", phoneNumber: "0123456789")); 
```

Comment:
Should create another class called 'MockContact' put it in '/mock_data' folder, responsible for generating mock data. Example:
`class MockContact { 
static List<Contact> mocks() => ...
}`